Client/Frontend/Settings/Brave Today/BraveTodayAddSourceViewController.swift

```diff
@@ -0,0 +1,371 @@
+        case dataTaskError(Error)
+        /// The data was either not received or is in the incorrect format
+        case invalidData
+        /// The data downloaded did not match a
+        case parserError(ParserError)
+        /// No feeds were found at the given URL
+        case noFeedsFound
+        
+        var localizedDescription: String {
+            switch self {
+            case .dataTaskError(let error as URLError) where error.code == .notConnectedToInternet:
+                return error.localizedDescription
+            case .dataTaskError:
+                return Strings.BraveToday.addSourceNetworkFailureMessage
+            case .invalidData, .parserError:
+                return Strings.BraveToday.addSourceInvalidDataMessage
+            case .noFeedsFound:
+                return Strings.BraveToday.addSourceNoFeedsFoundMessage
+            }
+        }
+    }
+    
+    private var pageTask: URLSessionDataTask?
+    private func downloadPageData(for url: URL, _ completion: @escaping (Result<[RSSFeedLocation], FindFeedsError>) -> Void) {
+        pageTask = session.dataTask(with: url) { [weak self] (data, response, error) in
+            guard let self = self else { return }
+            if let error = error as? URLError, error.code == .cancelled {
+                return
+            }
+            if let error = error {
+                completion(.failure(.dataTaskError(error)))
+                return
+            }
+            guard let data = data, let root = try? HTMLDocument(data: data) else {
+                completion(.failure(.invalidData))
+                return
+            }
+            
+            // Check if `data` is actually an RSS feed
+            if case .success(let feed) = FeedParser(data: data).parse() {
+                // User provided a direct feed
+                var title: String?
+                switch feed {
+                case .atom(let atom):
+                    title = atom.title
+                case .json(let json):
+                    title = json.title
+                case .rss(let rss):
+                    title = rss.title
+                }
+                completion(.success([.init(title: title, url: url)]))
+                return
+            }
+            
+            // Check if `data` is actually an OPML list
+            if let opml = OPMLParser.parse(data: data), !opml.outlines.isEmpty {
+                let locations = opml.outlines.compactMap(self.rssLocationFromOPMLOutline)
+                if !locations.isEmpty {
+                    completion(.success(locations))
+                    return
+                }
+            }
+            
+            // Ensure page is reloaded to final landing page before looking for
+            // favicons
+            var reloadUrl: URL?
+            for meta in root.xpath("//head/meta") {
+                if let refresh = meta["http-equiv"], refresh == "Refresh",
```

Comment:
would it be worth to lowercase `refresh` here for extra safety?

Reply:
I could, this is copy and pasted from favicon fetcher as well, want me to update this there too?

Reply:
I would do it in both places,
https://www.w3.org/TR/WCAG20-TECHS/H76.html here is uses small letter

Reply:
Do you want to add it or maybe run through @jumde to confirm whether we should lowercase it?
All my other comments are resolved, good job on that

Reply:
I will add it